src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/DelegateHelpers.cs

```diff
@@ -12,9 +12,33 @@ internal static class DelegateHelpers
     {
         private static object[] s_emptyObjectArray = Array.Empty<object>();
 
-        internal static object[] GetEmptyObjectArray()
+        private const int CacheLength = 16;
+        [ThreadStatic]
+        private static object[][] _arrayCache;
+
+        internal static object[] GetObjectArray(int length)
+        {
+            if (length == 0)
+                return s_emptyObjectArray;
+
+            object[] result = null!;
+            if (length < CacheLength)
+            {
+                _arrayCache ??= new object[CacheLength][];
+                result = _arrayCache[length];
```

Comment:
Nit: `_arrayCache[0]` is always unused. Maybe we could offset everything by one?

Reply:
I will do `-1`. It would also ensure that we do not accidentally recycle len-0 arrays.